apps/api/src/domain/models/bibliographic-reference/book-bibliographic-reference/commands/create-book-bibliographic-reference/create-book-bibliographic-reference.command-handler.ts

```diff
@@ -0,0 +1,108 @@
+import { CommandHandler } from '@coscrad/commands';
+import { Inject } from '@nestjs/common';
+import { InternalError } from '../../../../../../lib/errors/InternalError';
+import { RepositoryProvider } from '../../../../../../persistence/repositories/repository.provider';
+import { DTO } from '../../../../../../types/DTO';
+import { ResultOrError } from '../../../../../../types/ResultOrError';
+import { Valid } from '../../../../../domainModelValidators/Valid';
+import getInstanceFactoryForResource from '../../../../../factories/getInstanceFactoryForResource';
+import { IIdManager } from '../../../../../interfaces/id-manager.interface';
+import { IRepositoryForAggregate } from '../../../../../repositories/interfaces/repository-for-aggregate.interface';
+import { AggregateType } from '../../../../../types/AggregateType';
+import { InMemorySnapshot, ResourceType } from '../../../../../types/ResourceType';
+import buildInMemorySnapshot from '../../../../../utilities/buildInMemorySnapshot';
+import { BaseCreateCommandHandler } from '../../../../shared/command-handlers/base-create-command-handler';
+import { BaseEvent } from '../../../../shared/events/base-event.entity';
+import { validAggregateOrThrow } from '../../../../shared/functional';
+import BibliographicReferenceCreator from '../../../common/bibliographic-reference-creator.entity';
+import { BibliographicReferenceType } from '../../../types/BibliographicReferenceType';
+import { BookBibliographicReference } from '../../entities/book-bibliographic-reference.entity';
+import { BookBibliographicReferenceCreated } from './book-bibliographic-reference-created.event';
+import { CreateBookBibliographicReference } from './create-book-bibliographic-reference.command';
+
+@CommandHandler(CreateBookBibliographicReference)
+export class CreateBookBibliographicReferenceCommandHandler extends BaseCreateCommandHandler<BookBibliographicReference> {
+    protected repositoryForCommandsTargetAggregate: IRepositoryForAggregate<BookBibliographicReference>;
+
+    aggregateType = AggregateType.bibliographicReference;
+
+    constructor(
+        protected readonly repositoryProvider: RepositoryProvider,
+        @Inject('ID_MANAGER') protected readonly idManager: IIdManager
+    ) {
+        super(repositoryProvider, idManager);
+
+        this.repositoryForCommandsTargetAggregate = this.repositoryProvider.forResource(
+            ResourceType.bibliographicReference
```

Comment:
If we update the repository provider's API, we can have `forAggregate` and do this dynamically in the base class.